crates/builder/src/signer.rs

```diff
@@ -16,15 +16,34 @@ pub enum LocalOrAws {
 impl LocalOrAws {
     /// Load a privkey or AWS signer from environment variables.
     pub async fn load(key: &str, chain_id: Option<u64>) -> Result<Self, ConfigError> {
-        if let Ok(wallet) = load_wallet(key) {
+        if let Ok(wallet) = wallet(key) {
             Ok(LocalOrAws::Local(wallet))
         } else {
-            let signer = load_aws_signer(key, chain_id).await?;
+            let signer = aws_signer(key, chain_id).await?;
             Ok(LocalOrAws::Aws(signer))
         }
     }
 }
 
+/// Load the wallet from environment variables.
+///
+/// # Panics
+///
+/// Panics if the env var contents is not a valid secp256k1 private key.
+pub fn wallet(private_key: &str) -> Result<LocalWallet, ConfigError> {
```

Comment:
these 2 free functions should be in the `impl LocalOrAws` block

Reply:
https://github.com/init4tech/zenith-rs/pull/12/commits/dabe1c128b2bc6508209acff5cb4685636fcad94